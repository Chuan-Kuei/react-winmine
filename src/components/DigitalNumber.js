import React from "react";
import PropTypes from "prop-types";

import DigitalSign from "./DigitalSign.js";
import "./digitalNumber.css";
const DIGITAL = {
  0: "1111110",
  1: "0011000",
  2: "0110111",
  3: "0111101",
  4: "1011001",
  5: "1101101",
  6: "1101111",
  7: "1111000",
  8: "1111111",
  9: "1111101"
};
const DIGITAL_POSITION = [
  "left-top",
  "top",
  "right-top",
  "right-bottom",
  "bottom",
  "left-bottom",
  "middle"
];

const DigitalNumber = ({ value }) => {
  const numberSign = DIGITAL[value];

  return (
    <div styleName="container">
      {DIGITAL_POSITION.map((position, index) => {
        const isActive = !!+numberSign[index];
        return (
          <DigitalSign key={position} position={position} active={isActive} />
        );
      })}
    </div>
  );
};

DigitalNumber.defaultProps = {
  value: 0
};

DigitalNumber.propTypes = {
  value: PropTypes.number
};

export default DigitalNumber;
