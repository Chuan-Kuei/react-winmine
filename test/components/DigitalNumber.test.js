import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

import DigitalNumber from "../../src/components/DigitalNumber";

describe("DigitalNumber", () => {
  test("render correctly", () => {
    const wrapper = renderer.create(<DigitalNumber />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  test("props value is 0", () => {
    const wrapper = shallow(<DigitalNumber value={0} />);
    const expectResults = [
      { position: "left-top", active: true },
      { position: "top", active: true },
      { position: "right-top", active: true },
      { position: "right-bottom", active: true },
      { position: "bottom", active: true },
      { position: "left-bottom", active: true },
      { position: "middle", active: false }
    ];
    assertDigitalSign(wrapper, expectResults);
  });

  test("props value is 1", () => {
    const wrapper = shallow(<DigitalNumber value={1} />);
    const expectResults = [
      { position: "left-top", active: false },
      { position: "top", active: false },
      { position: "right-top", active: true },
      { position: "right-bottom", active: true },
      { position: "bottom", active: false },
      { position: "left-bottom", active: false },
      { position: "middle", active: false }
    ];
    assertDigitalSign(wrapper, expectResults);
  });

  test("props value is 2", () => {
    const wrapper = shallow(<DigitalNumber value={2} />);
    const expectResults = [
      { position: "left-top", active: false },
      { position: "top", active: true },
      { position: "right-top", active: true },
      { position: "right-bottom", active: false },
      { position: "bottom", active: true },
      { position: "left-bottom", active: true },
      { position: "middle", active: true }
    ];
    assertDigitalSign(wrapper, expectResults);
  });

  test("props value is 3", () => {
    const wrapper = shallow(<DigitalNumber value={3} />);
    const expectResults = [
      { position: "left-top", active: false },
      { position: "top", active: true },
      { position: "right-top", active: true },
      { position: "right-bottom", active: true },
      { position: "bottom", active: true },
      { position: "left-bottom", active: false },
      { position: "middle", active: true }
    ];
    assertDigitalSign(wrapper, expectResults);
  });

  test("props value is 4", () => {
    const wrapper = shallow(<DigitalNumber value={4} />);
    const expectResults = [
      { position: "left-top", active: true },
      { position: "top", active: false },
      { position: "right-top", active: true },
      { position: "right-bottom", active: true },
      { position: "bottom", active: false },
      { position: "left-bottom", active: false },
      { position: "middle", active: true }
    ];
    assertDigitalSign(wrapper, expectResults);
  });

  test("props value is 5", () => {
    const wrapper = shallow(<DigitalNumber value={5} />);
    const expectResults = [
      { position: "left-top", active: true },
      { position: "top", active: true },
      { position: "right-top", active: false },
      { position: "right-bottom", active: true },
      { position: "bottom", active: true },
      { position: "left-bottom", active: false },
      { position: "middle", active: true }
    ];
    assertDigitalSign(wrapper, expectResults);
  });

  test("props value is 6", () => {
    const wrapper = shallow(<DigitalNumber value={6} />);
    const expectResults = [
      { position: "left-top", active: true },
      { position: "top", active: true },
      { position: "right-top", active: false },
      { position: "right-bottom", active: true },
      { position: "bottom", active: true },
      { position: "left-bottom", active: true },
      { position: "middle", active: true }
    ];
    assertDigitalSign(wrapper, expectResults);
  });

  test("props value is 7", () => {
    const wrapper = shallow(<DigitalNumber value={7} />);
    const expectResults = [
      { position: "left-top", active: true },
      { position: "top", active: true },
      { position: "right-top", active: true },
      { position: "right-bottom", active: true },
      { position: "bottom", active: false },
      { position: "left-bottom", active: false },
      { position: "middle", active: false }
    ];
    assertDigitalSign(wrapper, expectResults);
  });

  test("props value is 8", () => {
    const wrapper = shallow(<DigitalNumber value={8} />);
    const expectResults = [
      { position: "left-top", active: true },
      { position: "top", active: true },
      { position: "right-top", active: true },
      { position: "right-bottom", active: true },
      { position: "bottom", active: true },
      { position: "left-bottom", active: true },
      { position: "middle", active: true }
    ];
    assertDigitalSign(wrapper, expectResults);
  });

  test("props value is 9", () => {
    const wrapper = shallow(<DigitalNumber value={9} />);
    const expectResults = [
      { position: "left-top", active: true },
      { position: "top", active: true },
      { position: "right-top", active: true },
      { position: "right-bottom", active: true },
      { position: "bottom", active: true },
      { position: "left-bottom", active: false },
      { position: "middle", active: true }
    ];
    assertDigitalSign(wrapper, expectResults);
  });

  const assertDigitalSign = (wrapper, expectResults) => {
    expect(wrapper.find("DigitalSign")).toHaveLength(7);
    expectResults.forEach((expectResult, index) => {
      const DigitalSign = wrapper.find("DigitalSign");
      expect(DigitalSign.at(index).props()).toEqual(expectResult);
    });
  };
});
