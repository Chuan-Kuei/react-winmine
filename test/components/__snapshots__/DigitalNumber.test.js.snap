// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DigitalNumber render correctly 1`] = `
<div
  className="container"
>
  <div
    className="container left-top vertical active"
  />
  <div
    className="container top horizontal active"
  />
  <div
    className="container right-top vertical active"
  />
  <div
    className="container right-bottom vertical active"
  />
  <div
    className="container bottom horizontal active"
  />
  <div
    className="container left-bottom vertical active"
  />
  <div
    className="container middle horizontal"
  />
</div>
`;
